src/components/v2/Table/index.tsx

```diff
@@ -0,0 +1,72 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState } from 'react';
+import TableMUI from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+import Head from './Head';
+import { useStyles } from './styles';
+
+interface ITableRowProps {
+  key: string;
+  render: () => React.ReactNode | string;
+  value: string | number | boolean;
+}
+
+interface ITableProps {
+  title: string;
+  data: ITableRowProps[][];
+  columns: { id: string; label: string; orderable: boolean }[];
+  fetchData?: (params: { order: { by: string; direction: 'asc' | 'desc' } }) => void;
+}
+
+export function Table({ columns, data, fetchData, title }: ITableProps) {
+  const styles = useStyles();
+  const [orderBy, setOrderBy] = useState<typeof columns[number]['id'] | undefined>(undefined);
+  const [order, setOrder] = useState<'asc' | 'desc' | undefined>('asc');
+  const [rows, setRows] = useState(data);
+  const onRequestOrder = (property: typeof columns[number]['id']) => {
+    let newOrder: 'asc' | 'desc' = 'asc';
+    if (property === orderBy) {
+      newOrder = order === 'asc' ? 'desc' : 'asc';
+    }
+    setOrderBy(property);
+    setOrder(newOrder);
+    if (fetchData) {
+      fetchData({ order: { by: property, direction: newOrder } });
+    } else {
+      // order in place
+      const newRows = [...rows];
+      newRows.sort((a, b) => {
+        const rowIndex: number = columns.findIndex(column => column.id === property);
+        if (a[rowIndex].value < b[rowIndex].value) {
+          return newOrder === 'asc' ? -1 : 1;
+        }
+        if (a[rowIndex].value > b[rowIndex].value) {
+          return newOrder === 'asc' ? 1 : -1;
+        }
+        return 0;
+      });
+      setRows(newRows);
+    }
+  };
+
+  return (
+    <TableContainer css={styles.tableContainer} component={Paper}>
+      <TableMUI css={styles.table} aria-label={title}>
+        <Head columns={columns} orderBy={orderBy} order={order} onRequestOrder={onRequestOrder} />
+        <TableBody>
+          {rows.map((row: ITableRowProps[]) => (
```

Comment:
Small thing: you don't need to specify the type of `row`. Since `rows` is already typed as `ITableRowProps[][]`, TypeScript will infer that the type of `row` is `ITableRowProps[]`.